### PR TITLE
Attempt adding mock service to container if not found

### PR DESCRIPTION
**Pull request by cnizzardini**
Ref: #16700

### CODE REVIEW

File: src/Core/TestSuite/ContainerStubTrait.php [hunk 1]
```diff
@@ -144,7 +145,12 @@ public function modifyContainer(EventInterface $event, ContainerInterface $conta
         }
         foreach ($this->containerServices as $key => $factory) {
             if ($container->has($key)) {
-                $container->extend($key)->setConcrete($factory);
+                try {
+                    $container->extend($key)->setConcrete($factory);
+                } catch (NotFoundException $e) {
+                    $container->add($key, $factory);
+                    $container->extend($key)->setConcrete($factory);
```
Comment (ndm2) [inline src/Core/TestSuite/ContainerStubTrait.php]: This line seems unnecessary, as the definition with the mock has already been added. And I guess a test for that would be nice too :)

**Comment by markstory** [comment]
This seems like something we could do in `4.x` as it isn't changing any public APIs.

File: src/Core/TestSuite/ContainerStubTrait.php (hunk 1 above)
Comment (cnizzardini) [reply src/Core/TestSuite/ContainerStubTrait.php]: Sorry @ndm2 I don't follow.

Comment (ndm2) [reply src/Core/TestSuite/ContainerStubTrait.php]: `$container->add($key, $factory);` adds the mock to the container, you don't have to extend that definition, it already has the required concrete attached.

**Comment by cnizzardini** [comment]
> This seems like something we could do in `4.x` as it isn't changing any public APIs.

@markstory what is the easiest way to change the base and only pull in my changes? My gitfu is weak.

**Comment by markstory** [comment]
> what is the easiest way to change the base and only pull in my changes? My gitfu is weak.

Don't worry about it. I can squash merge this into 4.next and then backport the squash.

**Pull request merged**